File: Arrays/#48 Rotate Image/main.js
function rotate(matrix) {
    const n = matrix.length;

    for (let i = 0; i < n; i++) {
        for (let j = i; j < n; j++) {
            [matrix[i][j], matrix[j][i]] = [matrix[j][i], matrix[i][j]];
        }
    }

    // Reverse each row
    for (let i = 0; i < n; i++) {
        matrix[i].reverse();
    }
}
//use [matrix[i][j], matrix[j][i]] = [matrix[j][i], matrix[i][j]];

File: Arrays/#48 Rotate Image/main.py
class Solution:
    def rotate(matrix):
        """
        Do not return anything, modify matrix in-place instead.
        """
        for i in range(len(matrix)):
            for j in range(i,len(matrix)):
                temp = matrix[i][j]
                matrix[i][j] = matrix[j][i]
                matrix[j][i] = temp
        for i in range(len(matrix)):
            matrix[i].reverse()

# make sure to add () in .reverse() method
# use matrix[i][j], matrix[j][i] = matrix[j][i], matrix[i][j]
